app/views/miq_request/_pre_prov.html.haml

```diff
@@ -36,18 +36,38 @@
             = h(row.name)
           - if @edit[:vm_headers].include?("image?")
             %td.text-nowrap
-              - if row.respond_to?(:image?)
+              - if row.respond_to?(:volume_template?)
+                = h(_("Volume"))
+              - elsif row.respond_to?(:volume_snapshot_template?)
+                = h(_("Volume Snapshot"))
+              - elsif row.respond_to?(:image?)
                 = h(row.image? ? _("Image") : _("Snapshot"))
+
           %td
-            = h(row.operating_system.try(:product_name))
+            - if row.respond_to?(:volume_template?) || row.respond_to?(:volume_snapshot_template?)
```

Comment:
This is repeating the same condition 5 times. And inside a template :-(

I would put the condition to the very beginning and maybe have 2 different templates, one with just the N/As

Also I's say that the title would deserve a helper function to be called.

Thx!

Reply:
Cool, will revise.